File: packages/typescript/tests/nss-vectors.test.ts
import fs from 'fs';
import path from 'path';

import { generateNss } from '../src/validators/nss';

type NssVector = {
  subdelegacion: string;
  registro_anio: string;
  nacimiento_anio: string;
  secuencial: string;
  nss: string;
};

const vectorsPath = path.resolve(__dirname, '../../shared-data/tests/nss_vectors.json');
const vectors: NssVector[] = JSON.parse(fs.readFileSync(vectorsPath, 'utf-8'));

describe('NSS shared vectors', () => {
  test.each(vectors)('matches %s', (vector) => {
    const nss = generateNss(
      vector.subdelegacion,
      vector.registro_anio,
      vector.nacimiento_anio,
      vector.secuencial
    );
    expect(nss).toBe(vector.nss);
  });
});

File: packages/dart/test/nss_vectors_test.dart
import 'dart:convert';
import 'dart:io';

import 'package:catalogmx/catalogmx.dart';
import 'package:test/test.dart';

void main() {
  test('NSS shared vectors', () {
    final file = File('../shared-data/tests/nss_vectors.json');
    final data = jsonDecode(file.readAsStringSync()) as List<dynamic>;

    for (final raw in data) {
      final vector = raw as Map<String, dynamic>;
      final nss = generateNSS(
        subdelegation: vector['subdelegacion'] as String,
        registrationYear: vector['registro_anio'] as String,
        birthYear: vector['nacimiento_anio'] as String,
        sequential: vector['secuencial'] as String,
      );
      expect(nss, vector['nss']);
    }
  });
}
